lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -926,16 +969,29 @@ class ParseableInterfaceModuleLoaderImpl {
     if (!prebuiltCacheDir.empty()) {
       llvm::SmallString<256> scratch;
       std::unique_ptr<llvm::MemoryBuffer> moduleBuffer;
-      auto path = computePrebuiltModulePath(scratch);
-      if (path) {
+      if (auto path = computePrebuiltModulePath(scratch)) {
         if (swiftModuleIsUpToDate(*path, deps, moduleBuffer)) {
           LLVM_DEBUG(llvm::dbgs() << "Found up-to-date prebuilt module at "
                                   << path->str() << "\n");
           return DiscoveredModule::prebuilt(*path, std::move(moduleBuffer));
         } else {
-          LLVM_DEBUG(llvm::dbgs() << "Found out-of-date prebuilt module at "
+          LLVM_DEBUG(llvm::dbgs() << "Found out-of-date prebuilt module at "		
                                   << path->str() << "\n");
         }
+      } else {
+        // Hack: deal with prebuilds of modules that still use the target-based
+        // names. This code is duplicated from above but there doesn't seem to
+        // be a great way to factor it out due to the early return.
```

Comment:
Would
```c++
if (!path)
  path = computeFallbackPrebuiltModulePath(scratch)
```
work?

Reply:
…

Yes, it would. I wrote this against a version that didn't check `fs::exists` when computing the path, and then hastily revised it once I saw that that got in. I'll change it to that.